File: java/jimmer-core/src/main/java/org/babyfish/jimmer/example/core/model/Author.java
package org.babyfish.jimmer.example.core.model;

import org.babyfish.jimmer.Immutable;

import java.time.LocalDateTime;
import java.util.List;

@Immutable
public interface Author {

    String name();

    LocalDateTime lastModifiedTime();

    List<Book> books();
}

File: kotlin/save-command-kt/src/main/kotlin/org/babyfish/jimmer/example/save/model/Author.kt
package org.babyfish.jimmer.example.save.model

import org.babyfish.jimmer.sql.*

@Entity
@KeyUniqueConstraint
interface Author {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Key
    val firstName: String

    @Key
    val lastName: String

    val gender: Gender

    @ManyToMany(mappedBy = "authors")
    val books: List<Book>
}